src/pages/Recovery/Recovery.tsx

```diff
@@ -0,0 +1,221 @@
+
+const getDangerousBIP44ParamsByChainId: GetDangerousBIP44ParamsByChainId = () => {
+  return {
+    [KnownChainIds.EthereumMainnet]: [EthereumChainAdapter.defaultBIP44Params],
+    [KnownChainIds.AvalancheMainnet]: [AvalancheChainAdapter.defaultBIP44Params],
+    [KnownChainIds.ThorchainMainnet]: [THORChainChainAdapter.defaultBIP44Params],
+    [KnownChainIds.CosmosMainnet]: [CosmosChainAdapter.defaultBIP44Params],
+    [KnownChainIds.OsmosisMainnet]: [OsmosisChainAdapter.defaultBIP44Params],
+    [KnownChainIds.BitcoinMainnet]: [
+      { ...BitcoinChainAdapter.defaultBIP44Params, purpose: 84 },
+      { ...BitcoinChainAdapter.defaultBIP44Params, purpose: 49 },
+      { ...BitcoinChainAdapter.defaultBIP44Params, purpose: 44 },
+    ],
+    [KnownChainIds.BitcoinCashMainnet]: [
+      { ...BitcoinCashChainAdapter.defaultBIP44Params, purpose: 84 },
+      { ...BitcoinCashChainAdapter.defaultBIP44Params, purpose: 49 },
+      { ...BitcoinCashChainAdapter.defaultBIP44Params, purpose: 44 },
+    ],
+    [KnownChainIds.LitecoinMainnet]: [
+      { ...LitecoinChainAdapter.defaultBIP44Params, purpose: 84 },
+      { ...LitecoinChainAdapter.defaultBIP44Params, purpose: 49 },
+      { ...LitecoinChainAdapter.defaultBIP44Params, purpose: 44 },
+    ],
+    [KnownChainIds.DogecoinMainnet]: [
+      { ...DogecoinChainAdapter.defaultBIP44Params, purpose: 84 },
+      { ...DogecoinChainAdapter.defaultBIP44Params, purpose: 49 },
+      { ...DogecoinChainAdapter.defaultBIP44Params, purpose: 44 },
+    ],
+  }
+}
+
+type GetDangerousAddressesArgs = {
+  wallet: HDWallet
+}
+
+type GetDangerousAddresses = (
+  args: GetDangerousAddressesArgs,
+) => Promise<GetDangerousAddressesReturn>
+type GetDangerousAddressesReturn = {
+  input: {
+    targetChainId: ChainId
+    chainName: string
+    bip44Params: BIP44Params
+    accountType: UtxoAccountType | undefined
+  }
+  output: {
+    pubkey: string
+    addresses: string[]
+  }
+}[]
+
+const getDangerousAddresses: GetDangerousAddresses = async ({ wallet }) => {
+  const dangerousBip44ParamsByChainId = getDangerousBIP44ParamsByChainId()
+
+  if (!supportsBTC(wallet)) {
+    throw new Error(`Recovery: wallet does not support bitcoin`)
+  }
+
+  const chainAdapterManager = getChainAdapterManager()
+
+  const result = []
+
+  for (const affectedChainId of affectedChainsIds) {
+    const maybeChainAdapter = chainAdapterManager.get(affectedChainId)
+    if (!maybeChainAdapter) throw new Error(`Recovery: no chain adapter for ${affectedChainId}`)
+    const adapter = maybeChainAdapter as unknown as UtxoBaseAdapter<UtxoChainId>
+    const chainName = adapter.getName()
+    for (const [bip44ParamChainName, wrongChainId] of Object.entries(KnownChainIds)) {
+      const bip44ParamsArray = dangerousBip44ParamsByChainId[wrongChainId]
+      for (const bip44Params of bip44ParamsArray) {
+        // only segwit native produces bc1-prefixed addresses
+        const accountType = UtxoAccountType.SegwitNative
```

Comment:
i previously iterated all of the `UtxoAccountType`'s, but it generates `1` or `3` prefixed (legacy or segwit respectively) addresses, which are not what we want